modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rule/TableModifyConverterRule.java

```diff
@@ -126,4 +138,34 @@ private static PhysicalNode createAggregate(IgniteTableModify tableModify, RelOp
 
         return new IgniteProject(cluster, outTrait.replace(IgniteDistributions.single()), sumAgg, projections, convertedRowType);
     }
+
+    /**
+     * To perform delete we need row with key fields only.
+     * But input distribution contains the indexes of the key columns according to the schema (i.e. for the full row).
+     * This method aligns the keys to their indexes so that a row containing only the key fields can be read.
+     *
+     * <pre>
```

Comment:
as discussed earlier - can you extend this java doc ?

Reply:
Expanded javadoc with more descriptive examples.